app/models/recipe.rb

```diff
@@ -1,3 +1,9 @@
 class Recipe < ApplicationRecord
    validates_presence_of :name, :complexity, :genre
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
+
+   def total_cost
+      self.ingredients.sum(:cost)
```

Comment:
Great model method and use of AR here!

Reply:
However, I don't see a corresponding model test. Be sure your models are always at 100% coverage - before writing a model method, write the test to be sure what AR you implement is correct!